expanderr.go

```diff
@@ -709,7 +756,18 @@ func main() {
 		o = f
 	}
 
-	if err := logic(o, &build.Default, posn); err != nil {
+	var (
+		templ *template.Template
+		err   error
```

Comment:
Please scope err as narrowly as possible, i.e. move “var err error” into the if *noErrTemplate conditional

Reply:
I feel like I'm missing something simple here.
```go
	var templ *template.Template
	if *noErrTemplate != "" {
->	templ, err = template.New("noErr").Parse(*noErrTemplate)
		if err != nil {
			log.Fatalf("error parsing no-error-template: %v", err)
		}
	}

	if err := logic(o, &build.Default, posn, templ); err != nil {
		log.Fatal(err)
	}
```
This gets me a `templ declared and not used`, so I used the above hack to make it happy. I'm blanking on what the 'right' thing to do here is.

Reply:
You’re probably using := instead of =.

My point was that the declaration of err should be within the conditional block:

```go
var templ *template.Template
	if *noErrTemplate != "" {
var err error
	templ, err = template.New("noErr").Parse(*noErrTemplate)
		if err != nil {
			log.Fatalf("error parsing no-error-template: %v", err)
		}
	}
```